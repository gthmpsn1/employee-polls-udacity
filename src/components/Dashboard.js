import { connect } from "react-redux";
import Nav from "./Nav";
import QuestionCard from "./QuestionCard";
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';

const Dashboard = (props) => {
    const questions = Object.values(props.questions);
    const users = props.users;
    questions.sort((a,b) => b.timestamp - a.timestamp);
    const authedUser = props.authedUser;

    const hasVoted = (question) => {
        if (question.optionOne.votes.includes(authedUser) || question.optionTwo.votes.includes(authedUser)) {
            return true;
        } else {
            return false;
        }
    }

    return (
        <div data-testid={'test-dashboard-component'} className="dashboard-container">
            <Nav />
            <h2>Dashboard</h2>
            <div className="dashboard">
                <div className="questions">
                    <Tabs>
                        <TabList>
                            <Tab>Unanswered</Tab>
                            <Tab>Answered</Tab>
                        </TabList>
                        <TabPanel>
                            <h2>Unanswered Polls</h2>
                            <div className="question-grid">
                                {questions.map((question) => (
                                    !hasVoted(question)
                                        && <QuestionCard key={question.id} question={question} author={users[question.author].name} authedUser={authedUser} />
                                ))}
                            </div>
                        </TabPanel>
                        <TabPanel>
                            <h2>Answered Polls</h2>
                            <div className="question-grid">
                                {questions.map((question) => (
                                    hasVoted(question)
                                        && <QuestionCard key={question.id} question={question} author={users[question.author].name} authedUser={authedUser} />
                                ))}
                            </div>
                        </TabPanel>
                    </Tabs>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = ({questions, users, authedUser}) => ({
    questions: questions,
    users: users,
    authedUser: authedUser
})

export default connect(mapStateToProps)(Dashboard);