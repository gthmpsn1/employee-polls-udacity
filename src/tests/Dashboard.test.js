/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import middleware from "../middleware";
import reducer from "../reducers";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import Dashboard from "../components/Dashboard";

describe("Render Dashboard", () => {
    let mockStore = createStore(reducer, middleware);

    it("will match the snapshot", async () => {
        let view = render(
            <MemoryRouter>
                <Provider store={ mockStore }>
                    <Dashboard />
                </Provider>
            </MemoryRouter>
        );

        await expect(view).toMatchSnapshot();
    });
});