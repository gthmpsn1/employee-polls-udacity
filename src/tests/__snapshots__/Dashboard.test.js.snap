// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Dashboard will match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dashboard-container"
        data-testid="test-dashboard-component"
      >
        <nav
          class="nav"
        >
          <h1>
            Employee Polls 
            <span>
              by Gabriel Thompson
            </span>
          </h1>
          <ul>
            <div>
              <li>
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  href="/leaderboard"
                >
                  Leaderboard
                </a>
              </li>
              <li>
                <a
                  href="/new-poll"
                >
                  New Poll
                </a>
              </li>
            </div>
            <div>
              <li
                class="user-avatar"
                style="background-image: url(undefined);"
              />
              <li>
                <a
                  data-testid="TEST"
                  href="/"
                >
                  Logout, 
                </a>
              </li>
            </div>
          </ul>
        </nav>
        <h2>
          Dashboard
        </h2>
        <div
          class="dashboard"
        >
          <div
            class="questions"
          >
            <div
              class="question-header"
            >
              Current Polls
            </div>
            <div
              class="legend"
            >
              <div
                class="legend-info"
              >
                <div
                  class="legend-unanswered"
                />
                <div
                  class="legend-item-name"
                >
                  = NOT VOTED
                </div>
              </div>
              <div
                class="legend-info"
              >
                <div
                  class="legend-answered"
                />
                <div
                  class="legend-item-name"
                >
                  = VOTED
                </div>
              </div>
            </div>
            <div
              class="question-grid"
            />
          </div>
          <div
            class="questions"
          >
            <div
              class="question-header"
            >
              Archived Polls
            </div>
            <p>
              Polls older than 120 days.
            </p>
            <div
              class="question-grid"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dashboard-container"
      data-testid="test-dashboard-component"
    >
      <nav
        class="nav"
      >
        <h1>
          Employee Polls 
          <span>
            by Gabriel Thompson
          </span>
        </h1>
        <ul>
          <div>
            <li>
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/leaderboard"
              >
                Leaderboard
              </a>
            </li>
            <li>
              <a
                href="/new-poll"
              >
                New Poll
              </a>
            </li>
          </div>
          <div>
            <li
              class="user-avatar"
              style="background-image: url(undefined);"
            />
            <li>
              <a
                data-testid="TEST"
                href="/"
              >
                Logout, 
              </a>
            </li>
          </div>
        </ul>
      </nav>
      <h2>
        Dashboard
      </h2>
      <div
        class="dashboard"
      >
        <div
          class="questions"
        >
          <div
            class="question-header"
          >
            Current Polls
          </div>
          <div
            class="legend"
          >
            <div
              class="legend-info"
            >
              <div
                class="legend-unanswered"
              />
              <div
                class="legend-item-name"
              >
                = NOT VOTED
              </div>
            </div>
            <div
              class="legend-info"
            >
              <div
                class="legend-answered"
              />
              <div
                class="legend-item-name"
              >
                = VOTED
              </div>
            </div>
          </div>
          <div
            class="question-grid"
          />
        </div>
        <div
          class="questions"
        >
          <div
            class="question-header"
          >
            Archived Polls
          </div>
          <p>
            Polls older than 120 days.
          </p>
          <div
            class="question-grid"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
